app/models/katello/host/subscription_facet.rb

```diff
@@ -34,6 +35,19 @@ def import_database_attributes(consumer_params)
         end
       end
 
+      def update_consumer_type(consumer_params)
+        if consumer_params.try(:[], 'type').try(:[], 'label') == 'hypervisor'
+          self.consumer_type = 'hypervisor'
+        elsif consumer_params.try(:[], 'facts').try(:[], 'virt.is_guest') == "true"
```

Comment:
I agree those are important queries to be available.  Eric's 2nd issue he is working on is to allow searching a guest by its hypervisor.  That would facilitate this search so you could do something like:

consumer_type = guest and set? hypervisor
or
consumer_type = guest and null? hypervisor